src/common/api/enviroments.ts

```diff
@@ -0,0 +1,164 @@
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+
+import { toast } from '../components/ui/use-toast';
+import useAxios from '../hooks/useAxios';
+import { Environment } from '../types/environment';
+
+export const useEnvironmentsQuery = ({
+	collectionId,
+}: {
+	collectionId?: string | undefined;
+}) => {
+	const axios = useAxios();
+
+	const query = useQuery<Environment[]>({
+		queryKey: ['environment', collectionId],
+		queryFn: async () => axios?.get('/enviroment').then((res) => res.data),
+	});
+
+	return query;
+};
+
+export const useEnvironmentQuery = ({ id }: { id?: string }) => {
+	const axios = useAxios();
+
+	const query = useQuery<Environment>({
+		queryKey: ['environment', id],
+		queryFn: async () =>
+			axios?.get(`/enviroment/${id}`).then((res) => res.data),
+		enabled: !!id,
+	});
+
+	return query;
+};
+
+export const useCreateEnvironmentMutation = ({
+	collectionId,
+}: {
+	collectionId?: string;
+}) => {
+	const queryClient = useQueryClient();
+	const axios = useAxios();
+
+	const mutation = useMutation({
+		mutationFn: async ({
+			name,
+			value,
+			collectionId,
+		}: {
+			name: string;
+			value: string;
+			collectionId: string;
+		}) =>
+			axios
+				?.post('/enviroment', { name, value, collectionId })
```

Comment:
this endpoint should be /environment should change endpoint from API and then here